src/main.py

```diff
@@ -144,50 +143,151 @@ def run(options: Options) -> int:
 
 
 def parse_flags(flags: List[str]) -> Options:
-    parser = argparse.ArgumentParser(description="Decompile MIPS assembly to C.")
-    parser.add_argument("filename", help="input filename")
-    parser.add_argument("function", help="function index or name", nargs="?")
-    parser.add_argument(
-        "--debug", dest="debug", help="print debug info", action="store_true"
+    parser = argparse.ArgumentParser(
+        description="Decompile MIPS assembly to C.",
+        usage="%(prog)s [--context C_FILE] [--emit-globals] [--valid-syntax] [-f FN ...] filename [filename ...]",
```

Comment:
IMO remove --valid-syntax, it's a niche use-case and flags like --void should be just as common

(also I wonder if --emit-globals should be made the default)